codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,237 @@ public TextHeaders clear() {
+            DefaultTextHeaders.super.set(name, commaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.set(name, convertAndCommaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.set(name, convertAndCommaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        private TextHeaders addConverted(CharSequence name, CharSequence convertedValue) {
+            CharSequence currentValue = DefaultTextHeaders.super.get(name);
+            if (currentValue == null) {
+                DefaultTextHeaders.super.add(name, convertedValue);
+            } else {
+                DefaultTextHeaders.super.set(name, commaSeparate(currentValue, convertedValue));
+            }
+            return DefaultTextHeaders.this;
+        }
+
+        private CharSequence convertAndCommaSeparate(Object... values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    convertAndQuote(singleValue, values[i]).append(", ");
+                }
+                convertAndQuote(singleValue, values[end]);
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence commaSeparate(CharSequence[] values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    quote(singleValue, values[i]).append(", ");
+                }
+                quote(singleValue, values[end]);
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence convertAndCommaSeparate(Iterable<?> values) {
+            StringBuilder sb = new StringBuilder();
+            Iterator<?> iterator = values.iterator();
+            if (iterator.hasNext()) {
+                Object next = iterator.next();
+                while (iterator.hasNext()) {
+                    convertAndQuote(sb, next).append(", ");
+                    next = iterator.next();
+                }
+                convertAndQuote(sb, next);
+            }
+            return sb.toString();
+        }
+
+        private CharSequence commaSeparate(Iterable<? extends CharSequence> values) {
```

Comment:
I'm trying to think of a way we can avoid the duplication between `commaSeparate` and `convertAndCommaSeparate`. Would it make sense to create a new interface to avoid duplicating this algorithm with `private CharSequence convertAndCommaSeparate(Iterable<?> values)`?  What about something like the following:

```java
private interface CsvValueEscaper<T> {
        StringBuilder escape(StringBuilder input, T value);
    }

    static final CsvValueEscaper<Object> ObjectEscaper = new CsvValueEscaper<Object>() {
        @Override
        public StringBuilder escape(StringBuilder input, Object value) {
            // use the convert method and then quote
        }
    };

    static final CsvValueEscaper<CharSequence> CharEscaper = new CsvValueEscaper<CharSequence>() {
        @Override
        public StringBuilder escape(StringBuilder input, CharSequence value) {
            // just quote
        }
    };

    private static <T> CharSequence commaSeparate(Iterable<? extends T> values, CsvValueEscaper<T> escaper) {
        StringBuilder sb = new StringBuilder();
        Iterator<? extends T> iterator = values.iterator();
        if (iterator.hasNext()) {
            T next = iterator.next();
            while (iterator.hasNext()) {
                escaper.escape(sb, next).append(", ");
                next = iterator.next();
            }
            escaper.escape(sb, next);
        }
        return sb.toString();
    }
```

Then to use the interface something like:

```java
Iterable<? extends CharSequence> values = null;
Iterable<?> values2 = null;
        
commaSeparate(values, CharEscaper);
commaSeparate(values2, ObjectEscaper);
```